doc/spec_TransferElement.dox

```diff
@@ -213,9 +232,50 @@ This documnent is currently still **INCOMPLETE**!
+  - 12.4 A TransferGroup is becomes read-only (cf. TransferGroup::isReadOnly()), if at least one of the member TransferElements is read-only.
+  - 12.5 Adding the same TransferElement twice is not allowed (logic_error).
+  - 12.6 Adding one TransferElement to two different TransferGroups is not allowed (logic_error).
+  - 12.7 Adding TransferElements with AccessMode::wait_for_new_data is not allowed (logic_error).
+  - 12.8 Adding a copy of the a TransferElement which is already in the group (returned by two different calls to Device::getZzzRegisterAccessor() resp. DeviceBackend::getRegisterAccessor() with the same arguments) will replace one of the identical TransferElement with a decorator which copies the data from the other.
+    - 12.8.1 This avoids an unnecessary transfer of identical data.
+    - 12.8.2 This is only possible, if the TransferElement is added by passing a TransferElementAbstractor to TransferGroup::addAccessor().
+  - 12.9 Operations executed via TransferGroup::read() and TransferGroup::write():
+    - 12.9.1 first call preXxx() on all high-level elements (cf. \ref transferElement_B_12_1_2_1 "12.1.2.1"),
+    - 12.9.2 next call the corresponding xxxTransferYyy() on all low-level elements (cf. \ref transferElement_B_12_1_2_2 "12.1.2.2"), and
+    - 12.9.3 finally call postXxx() on all high-level elements.
+  - 12.10 The TransferGroup implements a similar exception handling in TransferGroup::read() and TransferGroup::write() as their TransferElement counterparts. This especially means:
+    - 12.10.1 Before preXxx() is called, the TransferGroup checks all pre-conditions (device opened and operation allowed) to make sure no ChimeraTK::logic_error can occur (other exceptions are anyway not allowed in preXxx).
+      - \anchor transferElement_B_12_10_1_1 12.10.1.1 If any ChimeraTK::runtime_error occurs while checking the pre-conditions, it is immediately passed on to the application.
+      - \anchor transferElement_B_12_10_1_2 12.10.1.2 Any pre-condition which is not met will immediately result in a thrown ChimeraTK::logic_error
+      - 12.10.1.3 The exceptions from \ref transferElement_B_12_10_1_1 "12.10.1.1" and \ref transferElement_B_12_10_1_2 "12.10.1.2" are not prioritised by type. The first detected error throws.
+    - 12.10.2 ChimeraTK::runtime_error exceptions thrown by the low-level elements are propagated to the corresponding high-level elements so they are seen by their postXxx() functions (c.f. \ref transferElement_B_6_3 "6.3")
+    - 12.10.3 postXxx() is called for all high level elements, even if some postXxx() throw exceptions (c.f. \ref transferElement_B_5 "5").
+      - 12.10.3.1 If exceptions have been caught from any postXxx(), an exception is thrown after all of them have been executed.
+      - 12.10.3.2 If more than one exception type is detected, the exception type with the highest priority is thrown:
```

Comment:
Proposal:
```
* 12.10.3.2 As only either runtime_errors or boost::bad_numeric_cast can occur (see 12.11.1.1), it is clear which exception type to throw (see 12.11.1.1).
```

Reply:
Maybe remove that entire block. Ignore stupid cast errors. Then only runtime_errors are left. Put a comment that thread_interrupted cannot happen. That's it. Everything else is unnecessary bloat.

Reply:
Yes, that's even simpler. But with your proposal to only throw the first exception it will look like this: (not spec, what happens)

* execute transfer phase. Exceptions are delayed.
* call all postXxx() with their correct exceptions (and updateDataBuffer = false for all if any exception in transfer phase)
* catch all runtime_errors of the postXxx() phase and ignore them (logic and tread_interrupted cannot occur, numeric we don't care)
* Iterate all lowLevel TEs again and re-throw the first exception that is not nullptr.

Reply:
Again I am confused about the many discussions about the same (?) topic here. Is anything in this thread what has not been mentioned/clarified by the one about 12.10.3.1?

Reply:
Let's converge this discussion in the thread about 12.10.3.1.